plugins/statistics.koplugin/main.lua

```diff
@@ -172,6 +177,16 @@ function ReaderStatistics:addToMainMenu(tab_item_table)
                     })
                 end
             },
+            {
+                text = _("Time range"),
+                callback = function()
+                    total_msg, kv_pairs = self:getAdvStats()
```

Comment:
Looks like you are using keyvaluepage as a sub-menu (as you are not setting the value)? If so, it's better to create a sub-menu here:

```lua
{
    text = _("Time range"),
    sub_item_table = {
        {
            text = _("Last week"),
            callback = function()
                UIManager:show(KeyValuePage:new{
                    title = _("Last week"),
                    kv_pairs = getDatesFromAll(7, "daily_weekday"),
                })
            end,
        },
        {
            text = _("Last month by day"),
            callback = function()
                UIManager:show(KeyValuePage:new{
                    title = _("Last month by day"),
                    kv_pairs = getDatesFromAll(30, "daily_weekday"),
                })
            end,
        },
        {
            text = _("Last year by day"),
            callback = function()
                UIManager:show(KeyValuePage:new{
                    title = _("Last year by day"),
                    kv_pairs = getDatesFromAll(365, "daily"),
                })
            end,
        },
        {
            text = _("Last year by week"),
            callback = function()
                UIManager:show(KeyValuePage:new{
                    title = _("Last year by week"),
                    kv_pairs = getDatesFromAll(365, "weekly"),
                })
            end,
        },
        {
            text = _("Last 10 years by month"),
            callback = function()
                UIManager:show(KeyValuePage:new{
                    title = _("Last 10 years by month"),
                    kv_pairs = getDatesFromAll(3650, "monthly"), -- last 10 years
                })
            end,
        },
    }
}
```